bundles/org.openhab.binding.solax/src/main/resources/OH-INF/thing/localConnectInverter.xml

```diff
@@ -113,6 +113,24 @@
 				<description>The sum of PV currents from all strings</description>
 			</channel>
 
+			<channel id="inverter-temperature1" typeId="inverter-temperature">
+				<label>Inverter Temperature 1</label>
+				<description>Temperature 1 of the inverter</description>
+			</channel>
+			<channel id="inverter-temperature2" typeId="inverter-temperature">
+				<label>Inverter Temperature 2</label>
+				<description>Temperature 2 of the inverter</description>
+			</channel>
+			<channel id="inverter-workmode-code" typeId="inverter-workmode-code">
+				<label>Inverter Workmode Code</label>
+				<description>Inverter Workmode Code (0=Waiting, 1=Checking, 2=Normal, 3=Off, 4=Fault, 5=Permanent fault,
```

Comment:
You can add possible states with the option tags to support UI and i18n

Reply:
Hi, I prefer to keep that in the ENUM instead of the metadata unless it's strictly against the rules. 
There is a bigger change in my other PR (for the cloud connector) where I added Locale and timezone from the framework and my plan (already implemented there for the other relevant ENUMs) is to use the ENUM  value as an I18N key. Please let me know if this fits the requirement. I will refactor this one as a part of the other PR once this is merged (hopefully before the cloud connection PR)

Reply:
Not sure what your exact implementation would look like, but it sounds like a re-invention of base functionality. When this xml has the option tags, all translation is taken care of.  See the key/value pair enum/options here:
https://www.openhab.org/docs/developer/utils/i18n.html#thing-types

Reply:
Agree. I will make the translations as options and will pass on the ENUM as a value, hopefully the framework will do it's job here. Looks much more clean. Question is do we leave the code as a numeric code and I do the translation part for the non-code channel (where I create an ENUM from the code)?
This one should be the translated as a string status:
`<channel id="inverter-workmode" typeId="inverter-workmode">`

Reply:
One thing I didn't understand - if I add the options in the XML, do I always need to add them in the .properties file or it's assumed added and only the non-english translations must be added?
for example:
`<channel id="inverter-workmode" typeId="inverter-workmode">
   <options>
          <option value =  "NORMAL">Normal</option>
           .....
`
So do I have to add the following to the .properties file ? Something like this:
`thing-type.config.solax.inverter-workmode.language.option.NORMAL = Normal`

Reply:
They are needed in the `.properties` file in order to be translateable. You don't need to add them manually, you can run `mvn i18n:generate-default-translations` to update from the XML definitions - see https://www.openhab.org/docs/developer/utils/i18n.html#generating-i18n-properties-file

Reply:
Hi, I think I got it right after a few failed tests now. Tested it with my inverter and looks to be working. What I did is..

1. Removed the wormode-code channel from the metadata completely
2. Added options tot he thing type of the workmode channel. Agree that these numbers do not make sense, therefore the channel should be workmode and not workmode code anymore.